src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -442,7 +442,7 @@ public override ModelPropertyCollection Properties
                 if (_properties == null)
                 {
                     var properties = _provider.GetMetadataForProperties(ModelType);
-                    properties = properties.OrderBy(p => p.Order);
+                    properties = properties.OrderBy(p => p.Order).ThenBy(p => p.PropertyName);
```

Comment:
Agree this should have a specified comparer. Lean toward `Ordinal` as well though the order of this collection mostly affects generated HTML, where case is irrelevant.

Any votes for `OrdinalIgnoreCase`?